code/monitor/checkmonitor.q

```diff
@@ -123,6 +132,16 @@ addconfig:{
   `checkstatus upsert select checkid,family,metric,process,lastrun:0Np,nextrun:.z.p,status:0Nh,executiontime:0Nn,totaltime:0Nn,timerstatus:0Nh,running:0Nh,result:(count process)#() from x;  
  }
 
+copyconfig:{[checkid;newproc]
+  //function to copy config from checkconfig table and reinsert with new target process
+  //check if supplied checkid exists
+  if[not checkid in exec checkid from checkconfig;
+    '"supplied checkid doesn't exist in checkconfig table"
+  ];
+  newcheckid: 1i+exec max checkid from checkconfig;
+  `checkconfig upsert update checkid:newcheckid,process:newproc from 0!select from checkconfig where checkid=checkid
+ }
```

Comment:
If this is to duplicate a check from one process to another, I think you should pass it to addcheck or addconfig to make sure that it gets implemented properly and actually run (I think if you just insert it directly to checkconfig then that will not happen)